### PR TITLE
User can compose message

### PR DESCRIPTION
**Pull request by GergKllai1**
Creates tests so the user can compose message user story.

- As a background the user logs in and chooses the inbox where the compose message is
- Tests the happy path, where the user inputs all the fields and receives a success message 
- Tests the sad path, where user does not input all the fields and receives an error message
- Had to change the conversations_controller to include error message when no input was given.

PT:
https://www.pivotaltracker.com/n/projects/2226927

### CODE REVIEW

File: features/step_definitions/assertion_steps.rb [hunk 1]
```diff
@@ -6,6 +6,18 @@
     expect(current_path).to eq new_user_registration_path
 end
 
-When('I fill in {string} field with {string}') do |field, input|
-    fill_in field, with: input
-end
\ No newline at end of file
+Then('I should be on the login page') do
+    current_path.should == new_user_session_path
```
Comment (faraznaeem) [inline features/step_definitions/assertion_steps.rb]: Should is deprecated please use `expect(page).to etc`

Comment (GergKllai1) [reply features/step_definitions/assertion_steps.rb]: Done

File: features/user_can_compose_message.feature [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        Then I should be on the login page
+        Then I fill in 'Email' field with 'real@email.com'
+        And I fill in 'Password' field with 'password'
+        And I click 'Log in'
+        Then I should see 'Signed in successfully.'
+        And I should be on welcome site
+        Then I click 'Inbox'
+        And I should be on the inbox site
+        And I click 'Compose'
+        And I should be on the compose message site
```
Comment (faraznaeem) [inline features/user_can_compose_message.feature]: The background has way too many steps and needs to be refactored

try `And is logged in`

File: features/user_can_compose_message.feature [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+    Scenario: 'When I fill in the compose message form I can send a message to someone'
+        When I choose 'Greg' from 'Recipients'
+        And I fill in 'Subject' field with 'Hello'
+        And I fill in 'conversation_body' field with 'Hello there'
```
Comment (faraznaeem) [inline features/user_can_compose_message.feature:16]: `'conversation_body'` does not seem to be a good name for the field

File: features/user_can_compose_message.feature [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        And I click 'Message' 
+        Then I should see 'Your message was successfully sent!'
+        
+    Scenario: "When I don't fill in the compose message form I get an error message"
+        When I click 'Inbox'
```
Comment (faraznaeem) [inline features/user_can_compose_message.feature]: You are clicking on inbox both in the background and in the scenario

File: features/user_can_compose_message.feature [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        Then I should be on the inbox site
+        And I click 'Compose'
+        Then I should be on the compose message site
+        When I click 'Message' 
+        Then I should see 'You have to select a recipient!'
+        When I choose 'Greg' from 'Recipients'
+        And I click 'Message'
+        Then I should see 'You have to fill in subject and message!'
```
Comment (faraznaeem) [inline features/user_can_compose_message.feature:27]: The scenario needs to follow a better flow, you have multiple `When` `Then`

File: features/step_definitions/basic_steps.rb [hunk 6]
```diff
@@ -12,6 +12,19 @@
...
\ No newline at end of file
+end
+
+When('I choose {string} from {string}') do |input, field|
+    select input, from: field
+end
+
+#For debugging purposes
+#Given('show me the page') do
+#    save_and_open_page
```
Comment (faraznaeem) [inline features/step_definitions/basic_steps.rb]: code that is out commented = Shame

**Review by faraznaeem — CHANGES_REQUESTED** [review]

File: features/user_can_compose_message.feature (hunk 3 above)
Comment (GergKllai1) [reply features/user_can_compose_message.feature:16]: Messages text area is `conversation_body`, that is why we had to use it.

File: features/step_definitions/basic_steps.rb [hunk 7]
```diff
@@ -12,6 +12,22 @@
...
+
+When('I choose {string} from {string}') do |input, field|
+    select input, from: field
+end
+
+Given('I am logged in') do
+    visit new_user_session_path
+    fill_in 'Email', with: 'real@email.com'
+    fill_in 'Password', with: 'password'
+    click_on 'Log in'
```
Comment (magnus-thor) [inline features/step_definitions/basic_steps.rb]: You can use a warden helper here 
```
Given(/^I am logged in as "([^"]*)"$/) do |name|
  user = User.find_by(name: name)
  login_as(user, scope: :user)
end
```
Check this page for more information
https://class.craftacademy.co/courses/course-v1:CraftAcademy+CA-CC-01+2018/courseware/2c610fa3ac07421aa323ea6bbb010083/19ff5d7b9ba84a279f1d448fba0b1653/?activate_block_id=block-v1%3ACraftAcademy%2BCA-CC-01%2B2018%2Btype%40sequential%2Bblock%4019ff5d7b9ba84a279f1d448fba0b1653

File: features/step_definitions/basic_steps.rb [hunk 8]
```diff
@@ -12,6 +12,22 @@
...
+    visit new_conversation_path
```
Comment (magnus-thor) [inline features/step_definitions/basic_steps.rb:33]: Keep this in a separate step

**Review by magnus-thor — CHANGES_REQUESTED** [review]

**Review by magnus-thor — APPROVED** [review]
Good job

**Pull request merged**